### PR TITLE
Add macOS CI and use checkout@v2

### PR DESCRIPTION
**Pull request by gwynne**
Add macOS CI to the Github Actions workflow.

### CODE REVIEW

**Comment by Joannis** [comment]
@tanner0101 I don't think there's a need for a tag here, what would you prefer?

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -2,17 +2,25 @@ name: test
 on:
 - pull_request
 jobs:
+  vapor_macos:
+    runs-on: macos-latest
+    env:
+      DEVELOPER_DIR: /Applications/Xcode_11.4_beta.app/Contents/Developer
+    steps:
+    - uses: actions/checkout@v2
+    - run: brew install pkg-config libressl
```
Comment (0xTim) [inline .github/workflows/test.yml]: libressl isn't required anymore

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -2,17 +2,25 @@ name: test
...
+    - run: xcrun swift test --enable-test-discovery --sanitize=thread
```
Comment (0xTim) [inline .github/workflows/test.yml]: any reason we're not calling down to `swift` directly?

File: .github/workflows/test.yml (hunk 1 above)
Comment (gwynne) [reply .github/workflows/test.yml]: It refused to build until I added both formulae. Something to do with the weird setup on the runners would be my guess. 🤷‍♀

File: .github/workflows/test.yml (hunk 2 above)
Comment (gwynne) [reply .github/workflows/test.yml]: As far as using `xcrun` goes, in this case I was just making very doubly sure that the `DEVELOPER_DIR` override would take effect, at least for the `swift` invocation. (It's of course useless for affecting `brew`...)

File: .github/workflows/test.yml (hunk 1 above)
Comment (tanner0101) [reply .github/workflows/test.yml]: What error does it give if you omit `libressl`?

Comment (gwynne) [reply .github/workflows/test.yml]: It was complaining that it couldn't find `<openssl/conf.h>` while building NIO.

**Comment by gwynne** [comment]
Seems to be working fine without `libressl` now 🎉

**Comment by tanner0101** [comment]
Thanks!

**Pull request merged**